src/lib.rs

```diff
@@ -52,13 +74,15 @@ where
     let mut dest_image_file = image_file.clone();
 
     // create /tmp/{uuid}/ and copy image into
-    let mut tmp_image_file = PathBuf::from(format!("/tmp/{}", Uuid::new_v4()));
-    fs::create_dir_all(&tmp_image_file).context(format!(
+    let tmp_dir = PathBuf::from(format!("/tmp/{}", Uuid::new_v4()));
+    fs::create_dir_all(tmp_dir.clone()).context(format!(
         "run_image_command: couldn't create destination path {}",
-        tmp_image_file.to_str().unwrap()
+        tmp_dir.to_str().unwrap()
     ))?;
 
-    tmp_image_file.push(image_file.file_name().unwrap());
+    let _guard = TempDirGuard(tmp_dir.clone());
+
+    let mut tmp_image_file = tmp_dir.join(image_file.file_name().unwrap());
```

Comment:
only a remark: there is some inconsistency when we use `unwrap` and when `?`(e.g. line 64)

Reply:
fixed